analyzer/src/main/kotlin/AnalyzerResultBuilder.kt

```diff
@@ -58,46 +58,44 @@ class AnalyzerResultBuilder(private val curationProvider: PackageCurationProvide
             .resolvePackageManagerDependencies()
     }
 
-    fun addResult(projectAnalyzerResult: ProjectAnalyzerResult): AnalyzerResultBuilder {
-        // TODO: It might be, e.g. in the case of PIP "requirements.txt" projects, that different projects with
-        //       the same ID exist. We need to decide how to handle that case.
-        val existingProject = projects.find { it.id == projectAnalyzerResult.project.id }
-
-        if (existingProject != null) {
-            val existingDefinitionFileUrl = existingProject.let {
-                "${it.vcsProcessed.url}/${it.definitionFilePath}"
-            }
-            val incomingDefinitionFileUrl = projectAnalyzerResult.project.let {
-                "${it.vcsProcessed.url}/${it.definitionFilePath}"
-            }
+    fun addResult(projectAnalyzerResult: ProjectAnalyzerResult) = apply {
```

Comment:
@oss-review-toolkit/core-devs, so far we preferred to only start the implementation of a function expression after `=` if the whole implementation fits a single line. As that's not the case here, we would have formatted this like

```
fun addResult(projectAnalyzerResult: ProjectAnalyzerResult) =
    apply {
        // ...
    }
```

However, as recently @oheger-bosch and now also @fviernau used a syntax to start an opening block in the line of a function expression, I wonder whether we should officially endorse that style. The [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html#expression-bodies) are a bit vague on that. Opinions? Personally, I think I'd still prefer to add that line break, because otherwise it's easy to overlook the `apply` (in this case) as usually that line would only end with the return value.

Reply:
I've got no strong opinion on this. My proposal would be: 

1. put on a new line per default
2. In cases where much spaced / less indent is helping much, then put on same line (locally consistent).

So, in short, decide case by case.

Reply:
Outside of ORT I started using this style a lot to save indentation, especially in files where most functions are wrapped with some other function, like a database class where all functions are wrapped with a `transaction { ... }`. So I would be fine with adopting this style in ORT. It would be nice if there was a detekt rule for this.

Reply:
> It would be nice if there was a detekt rule for this.

I've [asked](https://kotlinlang.slack.com/archives/C88E12QH4/p1674638843243649).

> So, in short, decide case by case.

Ok, then let's at least allow the style.